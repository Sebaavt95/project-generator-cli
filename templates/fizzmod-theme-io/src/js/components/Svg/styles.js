import styled from 'styled-components';

const baseWrapper = styled.div`
	display: inline-flex;
	justify-content: center;
	align-items: center;
	width: ${({ size }) => `${size}px`};
	height: auto;
`;

const Icon = styled(baseWrapper)`
	color: ${({ color }) => color};
	& svg,
	& svg path,
	& svg polygon,
	& svg rect {
		fill: currentColor;
	}

	& svg {
		width: 100%;
		height: auto;
	}
`;
const Illustration = styled(baseWrapper)`
	& svg {
		width: 100%;
		height: auto;
	}
`;

Icon.displayName = 'Icon';
Illustration.displayName = 'Illustration';
export { Icon, Illustration };
