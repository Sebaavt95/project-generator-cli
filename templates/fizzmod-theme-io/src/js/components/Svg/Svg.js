/* eslint-disable react/no-danger */
/* eslint-disable global-require */
import React from 'react';
import { string, number, oneOf } from 'prop-types';
import WithErrorBoundary from 'components/Hocs/WithErrorBoundary';
import { Icon, Illustration } from './styles';

const Svg = ({ type, svgName, size, color, ...props }) => {
	if (!svgName) return null;

	const tryRequire = fileName => {
		try {
			// eslint-disable-next-line import/no-dynamic-require
			return require(`!!raw-loader!./svg/${fileName}.svg`);
		} catch (err) {
			return null;
		}
	};

	const svg = tryRequire(svgName) ? tryRequire(svgName).default : null;

	if (!svg) return null;

	if (type === 'illustration')
		return <Illustration {...props} size={size} dangerouslySetInnerHTML={{ __html: svg }} />;

	return <Icon {...props} size={size} color={color} dangerouslySetInnerHTML={{ __html: svg }} />;
};

Svg.propTypes = {
	/** Name of svg file  in svg folder */
	svgName: string,
	/** size of svg if is icon. If type is illustration, the size is width */
	size: number,
	color: string,
	type: oneOf(['icon', 'illustration'])
};

Svg.defaultProps = {
	svgName: '',
	size: 16,
	color: 'inherit',
	type: 'icon'
};

export default WithErrorBoundary(Svg);

// Only for Storybooks
export { Svg };
