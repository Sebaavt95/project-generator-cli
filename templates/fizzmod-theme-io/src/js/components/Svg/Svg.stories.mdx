import { Meta, Story, ArgsTable } from '@storybook/addon-docs/blocks';
import { Svg } from './Svg';

<Meta
	title="Componentes/Svg"
	component={Svg}
	argTypes={{
		color: { name: 'color', control: 'color' },
		svgName: {
			name: 'name Svg file',
			control: {
				type: 'select',
				options: ['newsletter', 'logo']
			}
		}
	}}
/>

# Svg

**Component for display Svgs**

## Usage

```
import Svg from 'components/Svg';
<Svg svgName="newsletter" size={28} type="icon"/>
```

## Props

<ArgsTable of={Svg} />

export const Template = args => <Svg {...args} />;

<Story name="Icon">
	<Svg svgName="newsletter" size={40} color="black" />
</Story>

<Story name="Illustrator">
	<Svg type="illustration" svgName="logo" size={120} />
</Story>

<Story name="Playground" args={{ svgName: 'newsletter', size: 40 }}>
	{Template.bind({})}
</Story>
