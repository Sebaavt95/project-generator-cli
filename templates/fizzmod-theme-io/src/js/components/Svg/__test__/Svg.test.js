import React from 'react';
import { shallow } from 'enzyme';

// eslint-disable-next-line import/no-named-as-default
import Svg from '../Svg';

describe('Svg component without svg name', () => {
	it('render with default props', () => {
		const wrapper = shallow(<Svg svgName="newsletter" />);
		const propColor = wrapper.props().color;
		const propType = wrapper.props().type;
		const propSize = wrapper.props().size;
		expect(propColor).toEqual('black');
		expect(propType).toEqual('icon');
		expect(propSize).toEqual(16);
	});
	it('render with size 28 and color white', () => {
		const wrapper = shallow(<Svg svgName="newsletter" size={28} color="#fff" />);
		const propColor = wrapper.props().color;
		const propType = wrapper.props().type;
		const propSize = wrapper.props().size;
		expect(propColor).toEqual('#fff');
		expect(propType).toEqual('icon');
		expect(propSize).toEqual(28);
		expect(wrapper).toMatchSnapshot();
	});
});
