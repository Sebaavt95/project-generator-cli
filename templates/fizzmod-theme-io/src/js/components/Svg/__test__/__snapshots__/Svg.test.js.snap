// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Svg component without svg name render with size 28 and color white 1`] = `
<Svg
  color="#fff"
  size={28}
  svgName="newsletter"
  type="icon"
/>
`;
